nestkernel/event.h

```diff
@@ -810,7 +814,19 @@ class CoeffArrayIterator
 class GapJEvent : public SecondaryEvent
 {
 private:
+  /*
+  Conceptual there is an one-to-one mapping between a SecondaryEvent
+  and a SecondaryConnectorModel. The synindex of this particular
+  SecondaryConnectorModel is stored in the static synid_ member variable
+  on model registration. There are however reasons (e.g. the usage of
+  CopyModel or the creation of the labeled synapse model duplicates for pyNN)
+  which make it necessary to register several SecondaryConnectorModels with
+  one SecondaryEvent. Therefore the synindices of all these models are
+  stored  in the synid_vector_. The has_syn_id()-function allows testing
+  if a particular synid is mapped with the SecondaryEvent in question.
+  */
   static synindex synid_;
+  static std::vector< synindex > synid_vector_;
```

Comment:
The only purpose of `synid_vector_` is to check whether the `syn_id` of a connection is in this vector. In general, `std::set< synindex >` would implement this more efficiently. But as long as this data structure is small, linear search through a vector might still be faster, so here we would need to think carefully about the relevant use cases.

In any case, the variable name should not contain the data structure. How about `supported_syn_ids_`?